src/crimson/tools/store-nbd.cc

```diff
@@ -168,6 +168,11 @@ struct request_context_t {
   seastar::future<> read_request(seastar::input_stream<char> &in) {
     return in.read_exactly(sizeof(struct nbd_request)
     ).then([this, &in](auto buf) {
+      if (buf.size() < sizeof(struct nbd_request)) {
```

Comment:
for posterity, `read_exactly()` returns an empty buffer in this case.

Reply:
In fact, it returns whatever is left in the channel, it may not be empty.

Reply:
thanks for pointing it out. you are right. `input_stream<CharType>::read_exactly_part()` fills `this->_buf` with the bits read from the stream and append them to the output and retry until it hits EOF.